mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -284,18 +350,35 @@ impl mayastor_server::Mayastor for MayastorSvc {
         self.locked(
             GrpcClientContext::new(&request, function_name!()),
             async move {
+
                 let rx = rpc_submit::<_, _, LvsError>(async move {
-                    Ok(ListPoolsReply {
-                        pools: Lvs::iter()
+                    Ok(
+                        Lvs::iter()
                             .map(|l| l.into())
                             .collect::<Vec<Pool>>(),
-                    })
+                    )
                 })?;
 
-                rx.await
-                    .map_err(|_| Status::cancelled("cancelled"))?
-                    .map_err(Status::from)
-                    .map(Response::new)
+                let rec  =  rx.await
+                        .map_err(|_| Status::cancelled("cancelled"))?
+                        .map_err(Status::from);
+
+                let mut lvs_pools = match rec {
+                    Ok(pools) => pools,
+                    Err(e) => return Err(e),
+                };
+
+                if  MayastorFeatures::get_features().lvm {
```

Comment:
Can we retain all found LVS pools and return the partial result to the user in case we see error upon LVM pools enumeration ? I would suggest just log the error.